src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -183,7 +179,9 @@ protected override void OnStopped()
             // the Disconnected event we subscribe is to detect #2 case. and this method is for #1 case. so when we are willingly disconnecting
             // we don't need the event, otherwise, Disconnected event will be called twice.
             UnregisterGlobalOperationNotifications();
-            UnregisterPersistentStorageLocationServiceChanges();
+
+            // Wait for any remaining tasks to be cleared, otherwise we might have OOP being torn down while we are still running
+            _currentRemoteWorkspaceNotificationTask.Wait();
```

Comment:
this line was in UnregisterPersistentStorageLocationServiceChanges.  i moved it here since i was removing the rest of the code in UnregisterPersistentStorageLocationServiceChanges.